### PR TITLE
Adds iterator

### PR DESCRIPTION
**Pull request by vieiralucas**
Aims to solve #14 

Check this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

### CODE REVIEW

**Comment by vieiralucas** [comment]
Well, babel is adding an `if (!true)` to the code, it's kinda hard to cover that.
What can we do about it?

**Comment by lucasfcosta** [comment]
WHOA!
That's something hard to cover, indeed.

I've gotta take a look into the generated code and see if there's some way of hinting Babel what it should do so we can cover it.
Would you mind waiting for me to take a look at it further this month? Maybe we will even need to take a look at Babel's code.

Oh, and thanks for the PR, awesome job!

How can we cover if(!true)?

**Comment by lucasfcosta** [comment]
Can you please rebase on `master` @vieiralucas?
I'm thinking about improving this with proxies and etcetera, it would be great to have this change.

I'll migrate every gulp task to NPM and fix this whole babel coverage thing.

File: src/harray.js [hunk 1]
```diff
@@ -151,6 +151,7 @@ Harray.prototype.get = function get(index) {
 };
 
 /**
+<<<<<<< 5bfc0631af9894e81d722878a8c8ade7242e0620
```
Comment (vieiralucas) [inline src/harray.js]: ooops

**Comment by vieiralucas** [comment]
Done

**Comment by vieiralucas** [comment]
Should I fix this?
https://travis-ci.org/lucasfcosta/harray/jobs/174356638#L315

**Comment by lucasfcosta** [comment]
Feel free to fix anything you want @vieiralucas!

**Comment by vieiralucas** [comment]
Well, I tried.

Apparentely we will need to run babel over the tests to support node 0.12.
Or remove all the const and lets from the tests.

IMHO, I think you should just use the `--compilers` flag with mocha when you switch to `npm scripts`.
I'm not willing to dig into how to do this with gulp for now.

**Comment by vieiralucas** [comment]
WTF? Buld passes?

**Comment by vieiralucas** [comment]
LOL
https://travis-ci.org/lucasfcosta/harray/jobs/174360462
gulp PLS

**Comment by lucasfcosta** [comment]
@vieiralucas feel free to move things to NPM if you want to, but I think that will be a pain...

**Comment by vieiralucas** [comment]
Can you merge this as it is?
We can tackle npm scripts in another PR.

**Comment by lucasfcosta** [comment]
@vieiralucas Yes I can! LGTM.

**Pull request merged**